### PR TITLE
feat: integrate Vite

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,24 +5,25 @@
...
+    "node": "~18.15.0"
   },
   "scripts": {
+    "preview": "vite preview",
     "analyze": "source-map-explorer 'build/static/js/*.js'",
-    "build": "yarn i18n && env-cmd -f .env react-app-rewired build",
-    "build-dev": "yarn i18n && env-cmd -f .env.dev react-app-rewired build",
-    "build-prod": "yarn i18n && env-cmd -f .env.production react-app-rewired build",
-    "build-stg": "yarn i18n && env-cmd -f .env.stg react-app-rewired build",
+    "build": "yarn i18n && tsc && env-cmd -f .env vite build",
```
Comment (hungdoansy) [inline package.json:13]: The order of tsc and vite build is different from the other commands

File: src/components/CurrencyInputPanel/CurrencyInputPanelBridge.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ interface CurrencyInputPanelBridgeProps {
   selectedChainId: ChainId | undefined
 }
 
-const noop = (value: string) => {
+const noop = (_value: string) => {
```
Comment (hungdoansy) [inline src/components/CurrencyInputPanel/CurrencyInputPanelBridge.tsx:38]: Can remove the params as well

File: src/pages/AddLiquidityV2/index.tsx [hunk 3]
```diff
@@ -576,7 +576,6 @@ export default function AddLiquidity() {
       tickUpper,
       pIndex,
       pool,
-      price,
```
Comment (hungdoansy) [inline src/pages/AddLiquidityV2/index.tsx:579]: Maybe delete by mistake?

File: src/pages/GrantProgram/EndedPrograms.tsx [hunk 4]
```diff
@@ -12,7 +12,6 @@ import { APP_PATHS } from 'constants/index'
 import useGetGrantPrograms from 'hooks/campaigns/useGetGrantPrograms'
 import useTheme from 'hooks/useTheme'
 import { MEDIA_WIDTHS } from 'theme'
-import { GrantProgram } from 'types/grantProgram'
```
Comment (hungdoansy) [inline src/pages/GrantProgram/EndedPrograms.tsx:15]: Weird. This type is used at line 95

File: src/pages/AddLiquidityV2/index.tsx (hunk 3 above)
Comment (namgold) [reply src/pages/AddLiquidityV2/index.tsx:579]: no, this params is unused var

File: package.json (hunk 1 above)
Comment (namgold) [reply package.json:13]: this one is correct

Comment (namgold) [reply package.json:13]: will fix those other

File: package.json [hunk 5]
```diff
@@ -148,25 +160,36 @@
     "@types/ua-parser-js": "^0.7.36",
     "@types/uuid": "^8.3.4",
     "@types/wcag-contrast": "^3.0.0",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
-    "@typescript-eslint/parser": "^4.33.0",
+    "@typescript-eslint/eslint-plugin": "^5.58.0",
+    "@typescript-eslint/parser": "^5.58.0",
+    "@vitejs/plugin-react": "^3.1.0",
+    "babel-plugin-macros": "^3.1.0",
     "compression-webpack-plugin": "^6.1.1",
```
Comment (hungdoansy) [inline package.json:167]: "react-app-rewire-styled-components"
"react-app-rewired"
"compression-webpack-plugin"
"source-map-explorer"

Please check these deps. They may not be in need any more

File: src/components/Icons/Solana.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import useTheme from 'hooks/useTheme'
 
-function Solana({ size }: { size?: number }) {
+function Solana({ _size }: { _size?: number }) {
```
Comment (viet-nv) [inline src/components/Icons/Solana.tsx:3]: LOL, why dont you remove it?

File: src/components/Icons/OptimismLogo.tsx [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import useTheme from 'hooks/useTheme'
 
-function OptimismLogo({ size }: { size?: number }) {
+function OptimismLogo({ _size }: { _size?: number }) {
```
Comment (viet-nv) [inline src/components/Icons/OptimismLogo.tsx:3]: we can remove this props

File: src/i18n.tsx [hunk 8]
```diff
@@ -19,9 +19,10 @@ const plurals: LocalePlural = {
 }
 
 async function dynamicActivate(locale: SupportedLocale) {
-  const { messages } = await import(`@lingui/loader!./locales/${locale}.po`)
+  const { messages } = await import(`./locales/${locale}.po`)
   i18n.loadLocaleData(locale, { plurals: plurals[locale] })
   i18n.load(locale, messages)
+  console.log('dynamicActivate final', { locale })
```
Comment (viet-nv) [inline src/i18n.tsx:25]: do we need this?

Comment (nguyenhoaidanh) [inline src/i18n.tsx:25]: rm

Comment (namgold) [reply src/i18n.tsx:25]: debug quên xoá ạ :shy:

**Review by viet-nv — APPROVED** [review]

File: .env.adpr [hunk 9]
```diff
@@ -0,0 +1,39 @@
+VITE_AGGREGATOR_API=https://meta-aggregator-api.kyberswap.com
```
Comment (nguyenhoaidanh) [inline .env.adpr:1]: revert to dev pls

**Pull request merged**